### PR TITLE
feat(uat): implement basic of API communication of paho-python client

### PR DESCRIPTION
**Pull request by eschastlivtsev**
**Description of changes:**
- Implement basic setup of Paho python client
- Implement helper scripts
- Implement Basic gRPC API communication functionality

**Why is this change necessary:**
- Part of Python Paho client implementation

**How was this change tested:**
Manual run of mqtt-client-control and python client.
MQTT functionality is not implemented, linking and shutdown steps are successful.

**Test results:**
Python Paho client output
```
python src/main.py test_id
asyncio: [DEBUG] Using selector: SelectSelector
GRPCLib: [INFO] Initialize gRPC library
GRPCLink: [INFO] Making gPRC client connection with 127.0.0.1:47619 as test_id...
GRPCLink: [INFO] Client connection with Control is established, local address is 127.0.0.1
grpc._cython.cygrpc: [DEBUG] Using AsyncIOEngine.POLLER as I/O engine
GRPCLink: [INFO] Handle gRPC requests
GRPCControlServer: [INFO] Server awaiting termination
GRPCControlServer: [INFO] CreateMqttConnection Placeholder TODO
GRPCControlServer: [INFO] shutdownAgent: reason That's it.
GRPCControlServer: [INFO] Server termination done
GRPCLink: [INFO] Shutdown gPRC link
Main: [INFO] Execution done successfully
```
Mqtt Client Control output
```
java -jar target/aws-greengrass-testing-mqtt-client-control.jar
WARNING: sun.reflect.Reflection.getCallerClass is not supported. This will impact performance.
[INFO ] 2023-06-05 18:47:41.333 [main] EngineControlImpl - MQTT client control gRPC server started, listening on 47619
[INFO ] 2023-06-05 18:47:44.631 [grpc-default-executor-0] GRPCDiscoveryServer - RegisterAgent: agentId test_id
[INFO ] 2023-06-05 18:47:44.653 [grpc-default-executor-0] GRPCDiscoveryServer - DiscoveryClient: agentId test_id address 127.0.0.1 port 64173
[INFO ] 2023-06-05 18:47:44.656 [grpc-default-executor-0] EngineControlImpl - Created new agent control for test_id on 127.0.0.1:64173
[INFO ] 2023-06-05 18:47:44.714 [grpc-default-executor-0] ExampleControl - Agent test_id is connected
[INFO ] 2023-06-05 18:47:44.721 [pool-2-thread-1] AgentTestScenario - Playing test scenario for agent id test_id
[ERROR] 2023-06-05 18:47:47.859 [pool-2-thread-1] AgentControlImpl - Couldn't create MQTT connection error '' CONNACK ''
[INFO ] 2023-06-05 18:47:47.866 [pool-2-thread-1] AgentControlImpl - shutdown request sent successfully
[INFO ] 2023-06-05 18:47:47.872 [grpc-default-executor-0] GRPCDiscoveryServer - UnregisterAgent: agentId test_id reason Agent shutdown by OTF request 'That's it.'
[INFO ] 2023-06-05 18:47:47.876 [grpc-default-executor-0] ExampleControl - Agent test_id is disconnected
```

**Any additional information or context required to review the change:**

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: uat/custom-components/client-python-paho/src/grpc_client_server/grpc_discovery_client.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        """
+        Construct GRPCDiscoveryClient
+        Parameters
+        ----------
+        agentId - id of agent to identify control channel by gRPC server
+        address - address of gRPC service, including port
+        """
+        self.__logger = GRPCDiscoveryClient.logger
+        self.__agent_id = agent_id
+        self.__channel = grpc.insecure_channel(address)
```
Comment (MikeDombo) [inline uat/custom-components/client-python-paho/src/grpc_client_server/grpc_discovery_client.py:34]: Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code assigns a resource that might leak to a class field, causing a resource leak that slows down or crashes your system. For example, when *mqtt\_client\_control\_pb2\_grpc.MqttAgentDiscoveryStub()* throws an exception, the *GRPCDiscoveryClient* object isn’t constructed so you can’t close this resource afterward.

**Fix**
Ensure that if an exception is thrown, the *\_\_channel* resource is closed before the constructor is exited.

Comment (eschastlivtsev) [reply uat/custom-components/client-python-paho/src/grpc_client_server/grpc_discovery_client.py:34]: fixed

Comment (bgklika) [reply uat/custom-components/client-python-paho/src/grpc_client_server/grpc_discovery_client.py:34]: UPD: fixed in d90f83c68c4d71b8a93f8b52631d3cc7aa67e472

File: uat/custom-components/client-python-paho/scripts/black_run.bat [hunk 2]
```diff
@@ -0,0 +1,7 @@
+@REM -------------------------------
+@REM Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+@REM SPDX-License-Identifier: Apache-2.0
+@REM -------------------------------
+
+@echo off
+black --line-length 79 --verbose --exclude="dev-env|grpc_client_server/grpc_generated" %~dp0\..\src
```
Comment (MikeDombo) [inline uat/custom-components/client-python-paho/scripts/black_run.bat:7]: please use 120 line length.

File: uat/custom-components/client-python-paho/scripts/lint_run.bat [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+@REM Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+@REM SPDX-License-Identifier: Apache-2.0
+@REM -------------------------------
+
+@echo off
+set PYTHONPATH=%~dp0..\src\
+set PYLINTRC=%~dp0..\src\.pylintrc
+pylint %~dp0..\src\
+flake8 --append-config %~dp0..\src\.flake8 %~dp0..\src\
+black --check --line-length 79 --exclude="dev-env|grpc_client_server/grpc_generated" %~dp0..\src\
```
Comment (MikeDombo) [inline uat/custom-components/client-python-paho/scripts/lint_run.bat:11]: 120

File: uat/custom-components/client-python-paho/scripts/lint_run.sh [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+
+DIR=$(realpath "$(dirname "$0")")
+export PYTHONPATH=$DIR/../src/
+export PYLINTRC=$DIR/../src/.pylintrc
+pylint $DIR/../src/
+flake8 --append-config $DIR/../src/.flake8 $DIR/../src/
+black --check --line-length 79 --exclude="dev-env|grpc_client_server/grpc_generated" $DIR/../src/
```
Comment (MikeDombo) [inline uat/custom-components/client-python-paho/scripts/lint_run.sh:13]: 120

File: uat/custom-components/client-python-paho/src/grpc_client_server/grpc_control_server.py [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+        request - incoming request
+        context - request context
+        Returns Empty object
+        """
+        self.__shutdown_reason = request.reason
+        self.__logger.info("shutdownAgent: reason %s", self.__shutdown_reason)
+        self.unblock_wait()
+        return Empty()
+
+    async def CreateMqttConnection(
```
Comment (MikeDombo) [inline uat/custom-components/client-python-paho/src/grpc_client_server/grpc_control_server.py:106]: why is this capitalized? Python functions should not be capitalized.

Comment (eschastlivtsev) [reply uat/custom-components/client-python-paho/src/grpc_client_server/grpc_control_server.py:106]: These capitalized functions are inherited from the GRPC-generated code (`from grpc_client_server.grpc_generated import mqtt_client_control_pb2_grpc`).

Generated code takes methods names from ".proto" file as is.
For example, in ".proto" we have 
`rpc CreateMqttConnection(MqttConnectRequest) returns (MqttConnectReply) {}`

The generated "mqtt_client_control_pb2_grpc.py" file code
`def CreateMqttConnection(self, request, context):`

We cannot change these method names to be aligned with the python code style.
In "grpc_control_server.py" I marked inherited overridden methods from generated code with "override" keyword in the methods headers.

File: uat/custom-components/client-python-paho/scripts/black_run.bat (hunk 2 above)
Comment (eschastlivtsev) [reply uat/custom-components/client-python-paho/scripts/black_run.bat]: Updated. Also added 120 line lengths for black and pylint.

File: uat/custom-components/client-python-paho/scripts/lint_run.sh (hunk 4 above)
Comment (eschastlivtsev) [reply uat/custom-components/client-python-paho/scripts/lint_run.sh]: Updated in53e0d06.
Also added 120 line lengths for black and pylint.

**Review by MikeDombo — COMMENTED** [review]
Create and use a pyproject.toml file instead of creating custom scripts for each OS.

Example pyproject file for using black
https://github.com/psf/black/blob/main/pyproject.toml


```toml
[tool.black]
line-length = 120

# required for compatibility with black:
profile = "black"

# To maintain consistency with other settings
line_length = 120
```

**Pull request merged**